### PR TITLE
AKS: clarify that one system pool is required

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This is a suggested improvement to the language that we print out when trying to set all node pools to `User`. The simplest way to run into this error is to build a single node pool AKS cluster, and then change the `mode` configuration from `System` to `User`.

I was a little confused with the existing error message:

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# azuremanagedmachinepools.infrastructure.cluster.x-k8s.io "pool0" was not valid:
# * Spec.Mode: Invalid value: "User": Last system node pool cannot be mutated to user node pool
#
```

I think this one might be a bit clearer:

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# azuremanagedmachinepools.infrastructure.cluster.x-k8s.io "pool0" was not valid:
# * Spec.Mode: Invalid value: "User": Cannot change node pool mode to User, you must have at least one System node pool in your cluster
#
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
I'd love for @LochanRn to weigh in on whether or not he things this is an improvement, as he authored the original feature:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1520/files

(belated thank you for that, btw!)

File: exp/api/v1beta1/azuremanagedmachinepool_webhook.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ func (m *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 			allErrs = append(allErrs, field.Invalid(
```
Comment (CecileRobertMichon) [inline exp/api/v1beta1/azuremanagedmachinepool_webhook.go:157]: I wonder if `Forbidden` would be a more appropriate error type here

> // Forbidden returns a *Error indicating "forbidden".  This is used to
// report valid (as per formatting rules) values which would be accepted under
// some conditions, but which are not permitted by current conditions (e.g.
// security policy).

Comment (jackfrancis) [reply exp/api/v1beta1/azuremanagedmachinepool_webhook.go:157]: I like it

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**